rest/form.go

```diff
@@ -0,0 +1,200 @@
+		Content: map[string]openapi3.MediaType{
+			"application/x-www-form-urlencoded": {
+				Schema: &schemaOrRef,
+			},
+		},
+	}
+
+	return openapi3.RequestBodyOrRef{
+		RequestBody: spec,
+	}, nil
+}
+
+// ReadRequest implements the [RequestReader] interface.
+func (fr *FormRequest[T]) ReadRequest(ctx context.Context, r *http.Request) error {
+	if err := r.ParseForm(); err != nil {
+		return BadRequestError{
+			Cause: err,
+		}
+	}
+
+	return decodeForm(r.Form, &fr.inner)
+}
+
+// Handle implements the [Handler] interface.
+func (h *ConsumeFormHandler[Req, Resp]) Handle(ctx context.Context, req *FormRequest[Req]) (*Resp, error) {
+	return h.inner.Handle(ctx, &req.inner)
+}
+
+// ConsumeOnlyForm creates a handler that consumes form requests without returning a response body.
+// Use this for webhook-style POST/PUT endpoints that process data but don't return content.
+//
+// Example:
+//
+//	c := rest.ConsumerFunc[Request](func(ctx context.Context, req *Request) error {
+//	    // process request
+//	    return nil
+//	})
+//	handler := rest.ConsumeOnlyForm(c)
+func ConsumeOnlyForm[T any](c Consumer[T]) *ConsumeFormHandler[T, EmptyResponse] {
+	inner := &ConsumerHandler[T]{
+		c: c,
+	}
+	return ConsumeForm(inner)
+}
+
+// HandleForm creates a handler that consumes form data and produces JSON responses.
+// Use this for POST/PUT form endpoints with JSON response bodies.
+//
+// Example:
+//
+//	h := rest.HandlerFunc[Request, Response](func(ctx context.Context, req *Request) (*Response, error) {
+//	    return &Response{Message: req.Message}, nil
+//	})
+//	handler := rest.HandleForm(h)
+func HandleForm[Req, Resp any](h Handler[Req, Resp]) *ConsumeFormHandler[Req, JsonResponse[Resp]] {
+	return ConsumeForm(ReturnJson(h))
+}
+
+// decodeForm decodes form values into a struct using reflection.
+// It looks for "form" struct tags to map form fields to struct fields.
+func decodeForm(form map[string][]string, dst any) error {
+	v := reflect.ValueOf(dst)
+	if v.Kind() != reflect.Ptr {
+		return errors.New("dst must be a pointer")
+	}
+
+	v = v.Elem()
+	if v.Kind() != reflect.Struct {
+		return errors.New("dst must be a pointer to a struct")
+	}
+
+	t := v.Type()
+	for i := 0; i < v.NumField(); i++ {
+		field := v.Field(i)
+		fieldType := t.Field(i)
+
+		// Skip unexported fields
+		if !field.CanSet() {
+			continue
+		}
+
+		// Get form field name from tag or use field name
+		formTag := fieldType.Tag.Get("form")
+		if formTag == "" {
+			formTag = strings.ToLower(fieldType.Name)
+		}
+
+		// Check if form value exists
+		values, ok := form[formTag]
+		if !ok || len(values) == 0 {
+			continue
+		}
+
+		// Set the field value based on its type
+		value := values[0]
+		if err := setField(field, value); err != nil {
+			return fmt.Errorf("failed to set field %s: %w", fieldType.Name, err)
+		}
+	}
+
+	return nil
+}
+
+// setField sets a reflect.Value to a string value, converting types as needed.
+func setField(field reflect.Value, value string) error {
```

Comment:
- Does this work with type aliased `string`, `int`, etc.?
- Does this work with `time.Time`, `time.Duration`?
- This should also allow the user type to implement `encoding.TextUnmarshaler`